torch/distributed/tensor/_api.py

```diff
@@ -329,7 +329,6 @@ def __tensor_unflatten__(inner_tensors, flatten_spec, outer_size, outer_stride):
         return DTensor(
             local_tensor,
             unflatten_spec,
-            requires_grad=requires_grad,
```

Comment:
You'll need to remove the `requires_grad` arg from the DTensor constructor. Although... that is technically BC breaking since DTensor is public API.

I would probably always make this False when called from `__tensor_unflatten__`, and rely on the fact that if we ever want our unflatten call to construct a DTensor that requires grad, we will be calling it from within an autograd context? (e.g. generating a DTensor graph output inside of the `CompileFunction(autograd.Function)` forward in AOTAutograd).